File: app/src/main/java/com/example/sarahn/waterreminderapp/ActCalculate.java
package com.example.sarahn.waterreminderapp;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.sarahn.waterreminderapp.Utils.Logging;
import com.example.sarahn.waterreminderapp.Utils.ToastLogger;
import com.example.sarahn.waterreminderapp.activities.ActTimeSpanChooser;
import com.example.sarahn.waterreminderapp.classes.BackpressedDialog;
import com.example.sarahn.waterreminderapp.classes.ClimateBottomSheet;
import com.example.sarahn.waterreminderapp.classes.ClsBottomSheetWeight;
import com.example.sarahn.waterreminderapp.classes.ClsBottomSheetActivity;
import com.example.sarahn.waterreminderapp.Utils.ClsRequirementCalculator;


public class ActCalculate extends Activity implements View.OnClickListener{

    public static TextView tvOne;
    public static TextView tvTwo;
    public static TextView tvThree;
    Button btnClaculate;


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_act_main);

        tvOne = (TextView) findViewById(R.id.tv_one);
        tvTwo = (TextView) findViewById(R.id.tv_two);
        tvThree = (TextView) findViewById(R.id.tv_three);
        btnClaculate = (Button) findViewById(R.id.btn_calculate);

        tvOne.setOnClickListener(this);
        tvTwo.setOnClickListener(this);
        tvThree.setOnClickListener(this);
        btnClaculate.setOnClickListener(this);
    }

    @Override
    public void onBackPressed() {
        BackpressedDialog.showDialog(this).show();
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()){
            case R.id.tv_one:
                ClsBottomSheetWeight bottomSheet = new ClsBottomSheetWeight(ActCalculate.this);
                bottomSheet.showDialog().show();
                //make the bellow line of code only works when it return from bottomsheet
               // tvOne.setText(Integer.toString(SharedPrefUtils.getWeight(this)));
                break;

            case R.id.tv_two:
                ClsBottomSheetActivity activityBottomSheet = new ClsBottomSheetActivity(ActCalculate.this);
                activityBottomSheet.showDialog().show();
                break;

            case R.id.tv_three:
                ClimateBottomSheet bottomSheetClimate = new ClimateBottomSheet(this);
                bottomSheetClimate.showDialog().show();
                break;

            case R.id.btn_calculate:
                ClsRequirementCalculator.calculateRequirement(ActCalculate.this);
                if(validateFeilds()){
                    Intent intent = new Intent(ActCalculate.this, ActTimeSpanChooser.class);
                    startActivity(intent);
                }else{
                    ToastLogger.toastMessage("please fill all feilds");
                }

                break;
        }

    }

    private boolean validateFeilds(){
        boolean isTrue;
        if(tvOne.getText().toString().matches("") ||
                tvTwo.getText().toString().matches("") ||
                tvThree.getText().toString().matches("")){
            Logging.logMessage("all true");
            isTrue = true;
        }else{
            isTrue = false;
        }
        return isTrue;
    }
}

File: app/src/main/java/com/example/sarahn/waterreminderapp/ActCalculate.kt
package com.example.sarahn.waterreminderapp

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import com.example.sarahn.waterreminderapp.ActCalculate
import com.example.sarahn.waterreminderapp.Utils.ToastLogger.toastMessage
import com.example.sarahn.waterreminderapp.activities.ActTimeSpanChooser
import com.example.sarahn.waterreminderapp.classes.ClimateBottomSheet
import com.example.sarahn.waterreminderapp.classes.ClsBottomSheetActivity
import com.example.sarahn.waterreminderapp.classes.ClsBottomSheetWeight
import com.example.sarahn.waterreminderapp.classes.ClsRequirementCalculator.calculateRequirement
import com.example.sarahn.waterreminderapp.dialogs.BackpressedDialog.showDialog

class ActCalculate : Activity(), View.OnClickListener {
    var btnClaculate: Button? = null
    var textViewOne: String? = null
    var textViewTwo: String? = null
    var textViewThree: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_act_main)
        tvOne = findViewById<View>(R.id.tv_one) as TextView
        tvTwo = findViewById<View>(R.id.tv_two) as TextView
        tvThree = findViewById<View>(R.id.tv_three) as TextView
        btnClaculate = findViewById<View>(R.id.btn_calculate) as Button
        tvOne!!.setOnClickListener(this)
        tvTwo!!.setOnClickListener(this)
        tvThree!!.setOnClickListener(this)
        btnClaculate!!.setOnClickListener(this)
    }

    override fun onBackPressed() {
        showDialog(this).show()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_one -> {
                val bottomSheet = ClsBottomSheetWeight(this@ActCalculate)
                bottomSheet.showDialog().show()
            }
            R.id.tv_two -> {
                val activityBottomSheet = ClsBottomSheetActivity(this@ActCalculate)
                activityBottomSheet.showDialog().show()
            }
            R.id.tv_three -> {
                val bottomSheetClimate = ClimateBottomSheet(this)
                bottomSheetClimate.showDialog().show()
            }
            R.id.btn_calculate -> {
                calculateRequirement(this@ActCalculate)
                if (validateFeilds()) {
                    val intent = Intent(this@ActCalculate, ActTimeSpanChooser::class.java)
                    startActivity(intent)
                } else {
                    toastMessage("please fill all feilds")
                }
            }
        }
    }

    private fun validateFeilds(): Boolean {
        return !(tvOne!!.text.toString().isEmpty() or
                tvTwo!!.text.toString().isEmpty() or
                tvThree!!.text.toString().isEmpty())
    }

    override fun onResume() {
        super.onResume()
        //        tvOne.setText(textViewOne);
//        tvTwo.setText(textViewTwo);
//        tvThree.setText(textViewThree);

//        tvOne.setText(tvOne.getText());
//        tvTwo.setText(tvTwo.getText());
//        tvThree.setText(tvThree.getText());
    }

    override fun onRestart() {
        super.onRestart()
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)


//        tvOne.setText(tvOne.getText());
//        tvTwo.setText(tvTwo.getText());
//        tvThree.setText(tvThree.getText());
        textViewOne = tvOne!!.text as String
        textViewTwo = tvTwo!!.text as String
        textViewThree = tvThree!!.text as String
        outState.putString(KEY_WEIGHT, textViewOne)
        outState.putString(KEY_AL, textViewTwo)
        outState.putString(KEY_CLIMATE, textViewThree)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)

//        savedInstanceState.getString(KEY_WEIGHT);
//        savedInstanceState.getString(KEY_AL);
//        savedInstanceState.getString(KEY_CLIMATE);
        tvOne!!.text = savedInstanceState.getString(KEY_WEIGHT)
        tvTwo!!.text = savedInstanceState.getString(KEY_AL)
        tvThree!!.text = savedInstanceState.getString(KEY_CLIMATE)
    }

    override fun onStop() {
        super.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    companion object {
        var tvOne: TextView? = null
        var tvTwo: TextView? = null
        var tvThree: TextView? = null
        const val KEY_WEIGHT = "weight"
        const val KEY_AL = "al"
        const val KEY_CLIMATE = "climate"
    }
}